crates/sui-config/src/node.rs

```diff
@@ -357,6 +365,10 @@ pub struct ConsensusConfig {
     pub submit_delay_step_override_millis: Option<u64>,
 
     pub narwhal_config: ConsensusParameters,
+
+    /// The choice of consensus protocol to run. We default to Narwhal.
+    #[serde(default = "default_consensus_protocol")]
+    pub protocol: ConsensusProtocol,
```

Comment:
Let me know if you believe that we need some additional guard to ensure that no one accidentally flips this

Reply:
Most other fields set the default at getter level, why is this field different?

Reply:
You are right on this. As those are public properties - and anyone can access them directly - I've avoided to use an `Option` as I wanted to make sure field is always initialised. Thus defaulting via a getter then didn't seem necessary.